sdk/formrecognizer/Azure.AI.FormRecognizer/src/CopyModelOperation.cs

```diff
@@ -45,7 +45,18 @@ public class CopyModelOperation : Operation<CustomFormModelInfo>
         public override string Id { get; }
 
         /// <inheritdoc/>
-        public override CustomFormModelInfo Value => OperationHelpers.GetValue(ref _value);
+        public override CustomFormModelInfo Value
+        {
+            get
+            {
+                if (HasCompleted && !HasValue)
```

Comment:
We have a race condition here.
```cs
if (update.Value.Status == OperationStatus.Succeeded)
{
  _hasCompleted = true;
  _value = ConvertValue(update.Value, _targetModelId, CustomFormModelStatus.Ready);
}
```
If you call this method between the attributions of `_hasCompleted` and `_value`, you'll get an `InvalidOperationException` with a `Copy model operation failed` message, which is not accurate.

Reply:
Maybe if you invert it:
```cs
if (update.Value.Status == OperationStatus.Succeeded)
{
  _value = ConvertValue(update.Value, _targetModelId, CustomFormModelStatus.Ready);
  _hasCompleted = true;
}
```
...?